### PR TITLE
bindi/Fixing no-such-table-error (WIP - for feedback)  

### LINKED ISSUE #30: 🐛 Bug: `NoSuchTableError` is raised if user tries to access a stream that did not receive records
Currently we only run 'create table' when and if record batches are being processed for that stream. Since we know which stream names are selected, and since we have the schema declared in the catalog for those streams, we could optionally _also_ create empty destination tables and initialize empty CachedDatasets for those streams.

This would be helpful because a user does not expect to get an error when checking for the results of a specific stream. Example, the "deployments" stream of the GitHub source is very likely to be empty. A user iterating through the streams they selected will get a confusing failure rather than just seeing an empty dataset.

Example error from Github (internal slack thread [here](https://airbytehq-team.slack.com/archives/C05DACCP6PM/p1707611362503629?thread_ts=1707603523.544589&cid=C05DACCP6PM)):
```
sqlalchemy.exc.NoSuchTableError: deployments
```

### PR DESCRIPTION
**Pull request by bindipankhudi**
Resolves: https://github.com/airbytehq/airbyte-lib-private-beta/issues/30

This is a WIP. Would love any feedback on changes and direction of work. 

**Main updates** 
- _processors.py: Added a Check for empty data streams in the process_input_stream method (This is the first way we talked about. Haven't explored the other options yet) 
- test_source_test_fixture.py: Added an empty stream (stream3) to source-test fixtuure and updated relevant tests. Everything is passing except for three tests I have disabled (more on that below)  

**Note:** 
- You will see three tests (which connect to PostGres, and snowflake) are disabled. This is because my terminal is unable to connect to postgres container (even though it's initialized properly). 
- Still have to run whole suite of tests, linting etc.. Once I resolve the above will continue with running other tests.

### CODE REVIEW

File: airbyte-lib/tests/integration_tests/fixtures/source-test/source_test/run.py [hunk 1]
```diff
@@ -35,6 +35,20 @@
                     },
                 },
             },
+            {
+                "name": "stream3",
+                "supported_sync_modes": ["full_refresh", "incremental"],
+                "source_defined_cursor": False,
+                "json_schema": {
```
Comment (aaronsteers) [inline airbyte-lib/tests/integration_tests/fixtures/source-test/source_test/run.py:42]: If this is a test stream that is intended to always be empty, what do you think of us changing the name and adding a description.

```suggestion
            {
                "name": "always-empty-stream",
                "description": "This stream always emits zero records, to test handling of empty datasets."
                "supported_sync_modes": ["full_refresh", "incremental"],
                "source_defined_cursor": False,
                "json_schema": {
```

Comment (aaronsteers) [reply airbyte-lib/tests/integration_tests/fixtures/source-test/source_test/run.py:42]: It does break the 'stream1', 'stream2', ... convention, so I guess I could see a case for going either way.

File: airbyte-lib/airbyte_lib/_processors.py [hunk 2]
```diff
@@ -203,12 +206,18 @@ def process_airbyte_messages(
...
+
         # We are at the end of the stream. Process whatever else is queued.
         for stream_name, stream_batch in stream_batches.items():
-            if stream_batch:
-                record_batch = pa.Table.from_pylist(stream_batch)
-                self._process_batch(stream_name, record_batch)
-                progress.log_batch_written(stream_name, len(stream_batch))
+            record_batch = pa.Table.from_pylist(stream_batch)
+            self._process_batch(stream_name, record_batch)
+            progress.log_batch_written(stream_name, len(stream_batch))
```
Comment (aaronsteers) [inline airbyte-lib/airbyte_lib/_processors.py:220]: This block makes sense to me. If we run into issues with the processor not being able to handle empty batches, we could evaluate other options - but this seems sound, well commented, and not overly complex.

File: airbyte-lib/airbyte_lib/_processors.py [hunk 3]
```diff
@@ -109,6 +111,7 @@ def register_source(
             incoming_source_catalog=incoming_source_catalog,
             incoming_stream_names=stream_names,
         )
+        self._expected_streams = stream_names
```
Comment (aaronsteers) [inline airbyte-lib/airbyte_lib/_processors.py:114]: Nice 👍

File: airbyte-lib/tests/integration_tests/fixtures/source-test/source_test/run.py (hunk 1 above)
Comment (bindipankhudi) [reply airbyte-lib/tests/integration_tests/fixtures/source-test/source_test/run.py:42]: yep, makes sense, so we don't accidentally add records for it later. will change. will add comments explaining that too.,

**Comment by bindipankhudi** [comment]
Did rebase without thinking. The branch changes got too hairy, so moves changes to a new branch, closing this one.

**Pull request closed without merging**